src/wp-includes/option.php

```diff
@@ -678,7 +674,7 @@ function add_option( $option, $value = '', $deprecated = '', $autoload = 'yes' )
 	}
 
 	$serialized_value = maybe_serialize( $value );
-	$autoload         = ( 'no' === $autoload || false === $autoload ) ? 'no' : 'yes';
+	$autoload         = get_autoload_value( $autoload, $option, $value );
```

Comment:
See my other comment, we should probably have a separate utility function that only determines the value to use when no autoload value is given. The current way this is implemented mixes up the concerns of determining the value with getting the value from the DB, which for example here is completely unnecessary, as we should be able to rely purely on the `$autoload` parameter here.